cscs-checks/libraries/boost/boost_python_check.py

```diff
@@ -12,8 +12,13 @@ class BoostPythonBindingsTest(rfm.RegressionTest):
     def __init__(self):
         self.descr = f'Test for Boost with Python bindings'
         self.valid_systems = ['daint:gpu', 'daint:mc', 'dom:gpu', 'dom:mc',
-                              'eiger:mc']
-        self.valid_prog_environs = ['builtin']
+                              'eiger:mc', 'pilatus:mc']
+
+        if self.current_system.name in ['eiger', 'pilatus']:
```

Comment:
Isn't the Boost module load its dependencies, like the `cpeGNU`?

Reply:
Or is it related to the hierarchical naming scheme?

Reply:
`cpeGNU` must be loaded first to be able to load `Boost`, due to the HMNS: that applies to every application built with a toolchain.